File: backend/app/schemas/employee.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime
from .department import DepartmentBase

class EmployeeBase(BaseModel):
    first_name: str
    last_name: str
    email: Optional[str]
    position: Optional[str]
    hire_date: Optional[datetime] = None
    phone: Optional[str] = None
    salary: Optional[str] = None
    status: Optional[str] = "active"
    checkin: Optional[str] = None
    checkout: Optional[str] = None

class EmployeeCreate(EmployeeBase):
    password: str
    department_id: Optional[int] = None 

class EmployeeLogin(BaseModel):
    email: str
    password: str

class TokenResponse(BaseModel):
    access_token: str
    refresh_token:str

class EmployeeOut(EmployeeBase):
    id: int
    department_name: Optional[str] = None


class Config:
    orm_mode = True

File: frontend/src/types/employee.ts
// export interface Employee {
//   id: string;
//   firstName: string;
//   lastName: string;
//   email: string;
//   position: string;
//   department: string;
//   checkin: string;
//   checkout: string;
//   phone?: string;
//   avatar?: string;
//   hireDate: string;
//   status: 'active' | 'inactive';
//   salary?: number;
//   rollNo: string;
//   password?: string;
// }

export interface Employee {
  id: number | string;
  firstName: string;
  lastName: string;
  email?: string;
  phone?: string;
  position?: string;
  status?: string;
  checkin?: string;
  checkout?: string;
  hireDate?: string;
  salary?: number;
  department?: string; // department_name from backend
  rollNo?: string; // optional if not from backend
}


export interface EmployeeFilters {
  department?: string;
  status?: 'active' | 'inactive';
  search?: string;
}